### DISCUSSION TITLE
Dynamic figure size

### DISCUSSION THREAD

**Discussion by slamao** [quarto-dev/quarto-cli#4028] [Q&A] [answered]
Hi, 
thank you very much for this great tool. 

I would like to ask if it is possible to set a dynamic figure size?
You can do this easily in RMarkdown (see below what I mean) but I cannot find a way how to do that in Quarto (using knitr). 

```
```{r}
my_width <- 10
```

```
```{r fig.width=my_width}
<plot code>
```

Any chances this works (or will work) in Quarto as well?

Many thanks!

**Comment by cderv** [comment, marked as answer]
Quarto does not support only YAML values for its chunk definition. However, for **knitr** compatibility, there is special support 
See https://quarto.org/docs/computations/r.html#chunk-options

* First you could still pass option as you do in R Markdown. So about "Any chances this works (or will work) in Quarto as well?" Did you try it ? Because it should work

* You can use YAML new syntax, but use special `!expr` syntax as in the doc - Only drawback: It applies on **knitr** only within Quarto. Not other computation engine. 

Hope it helps

**Reply by slamao** [reply]
Thanks Christophe, amazing, this works!! Thanks a lot!

To summarise for any future reference, use this in quarto to make your plot size dynamic:
```
```{r}
#| fig-width: !expr my_width
<plot code>
```
